src/ngraph_utils.cc

```diff
@@ -33,15 +34,67 @@
 #include "tensorflow/core/platform/default/logging.h"
 #include "tensorflow/core/platform/protobuf.h"
 
+#include "version.h"
+
 using namespace std;
 namespace ng = ngraph;
 
 namespace tensorflow {
 
 namespace ngraph_bridge {
 
+Status IsCopyLogEnabled(int graph_id, bool& is_copy_log_enabled) {
+  const char* copy_env_var = std::getenv("NGRAPH_TF_LOG_COPIES");
+  if (copy_env_var == nullptr) {
+    is_copy_log_enabled = false;
+    return Status::OK();
+  }
+  int test_graph_id;
+  try {
+    test_graph_id = stoi(string(copy_env_var));
+  } catch (const std::invalid_argument& ia) {
+    return errors::InvalidArgument("Invalid argument for NGRAPH_TF_LOG_COPIES");
+  }
+  // if -1 copies are logged for all graphs
+  is_copy_log_enabled = (test_graph_id == -1 || test_graph_id == graph_id);
+  return Status::OK();
+}
+void PrintTFTensor(Tensor& T1) {
+  NGRAPH_VLOG(4) << "all tensor values" << (T1).SummarizeValue(64) << endl;
+}
+std::string DebugNode(Node* node) {
+  std::string temp = node->name();
+  temp += "[" + node->type_string() + "]";
+  return temp;
+}
+
+std::string PrintBool(bool var) { return (var ? "Yes" : "No"); }
+
 bool IsNGVariableType(string node_type) {
-  return node_type == "NGraphVariable";
+  if (ngraph_tf_are_variables_enabled())
+    return (node_type == "NGraphVariable" || node_type == "NGraphAssign");
+  else
+    return node_type == "NGraphVariable";
+}
+
+bool IsNGSupportedType(string node_type) {
```

Comment:
Suggest `IsSupportedNGNodeType`

Reply:
This we can revisit on Monday, as we have to figure out a corresponding name for IsNGVariableType.